src/main/java/xyz/nkomarn/harbor/util/PlayerManager.java

```diff
@@ -99,7 +104,24 @@ public void updateActivity(@NotNull Player player) {
      * Registers Harbor's fallback listeners for AFK detection if Essentials is not present.
      */
     public void registerFallbackListeners() {
-        harbor.getServer().getPluginManager().registerEvents(new AfkListeners(), harbor);
+        if (afkListeners == null) {
+            afkListeners = new AfkListeners();
+            harbor.getServer().getPluginManager().registerEvents(afkListeners, harbor);
+        }
+    }
+
+    /**
+     * Unregisters Harbor's fallback listeners for AFK detection; can be used by
+     * an external plugin that is using an {@link AFKProvider} in order to prevent
+     * conflicts
+     */
+    public void unregisterFallbackListeners() {
+        if (afkListeners != null) {
+            for (HandlerList handlerList : HandlerList.getHandlerLists()) {
+                handlerList.unregister(afkListeners);
+            }
```

Comment:
I fully understand and will work on fixing the way the AFK registration works, and I can remove stafffacilities. Is there something you recommend using instead as an exclusion example? I simply used StaffFacilities because it's something my server is using, and it was an obvious example I could think of. I do like to have examples of API usage personally, so I'm more than happy to remove that if you can think of a good replacement.

I also was planning on adding a config option to disable fallback listeners from loading at all.


For some reason I didn't see an unregisterAll, I'll make the change asap

Reply:
I'd also be happy to pull in things from #87 if you'd like, or you can rebase those on top of my code

Reply:
Hooking in here.
A cleaner implementation would be to override or register an own afk provider from outside the plugin.

Avoiding dependencies for such a specific use case would be great.
My proposal would be:

Use the AFKProvider interface.
Implement your own afk listener in your Staff plugin.
Add a method in Harbor to register your own afk provider, which can either override all current registered afk providers or be used as an additional provider.
This would be way more flexible in many ways.
You could also add a method to disable the current AFK listener again in runtime.

This would avoid overhead for specific plugins and every plugin can integrate itself into harbor without the hassle of a pull request.

Reply:
Sounds good, does this mean leaving essentials support in, or making it a separate plugin/module

Reply:
Since essentials is the most common plugin I would leave it in there to provide a solid solution for the most cases. Also it is very unlikely that essentials will accept a pr to be compatible with harbor.

Reply:
TBH same for stafffacilities, so there would need to be a glue plugin. I'll maybe have them added as a hashmap, and a config section for allowed/enabled ones?

Reply:
For exclusion, as an example, I can actually rewrite it so that all exclusion providers (which will only be ORed) currently implemented are via the api, that could work well

Reply:
I think that we would need to differentiate.

Essentials has thousands of active users and has a high relevance in the overall community. Nearly everyone knows it.
StaffFacilitities has around 600 total downloads on spigot. I dont want to offend you, but that's not popular.

In general the responsibility should be on the provider side and data should be pushed into the plugin rather than pulled from all sources which may exist.
Its just that essentials is a special case since the availability of this source is given in a lot of cases.
Bit in the end this is just my opinion :D In the end I am not a maintainer xD Just a random dude who loves to code xD

Reply:
> For exclusion, as an example, I can actually rewrite it so that all exclusion providers (which will only be ORed) currently implemented are via the api, that could work well

The exclusions should be added as well like the afk providers I think yes. That would be way cleaner. The responsibility is on the provider side again.

Reply:
No offence taken! I was just trying to think of some way to, in source code, show how one could use the exclusions API. I could use Supervanish as an example, since that has ~95k downloads, but for now I think I'll use the provider method for internal functions, and that should be a good enough example!